nio/schemas.py

```diff
@@ -815,6 +815,10 @@ class Schemas:
                     "creator": {"type": "string", "format": "user_id"},
                     "m.federate": {"type": "boolean", "default": True},
                     "room_version": {"type": "string", "default": "1"},
+                    "predecessor": {
+                        "type": "array",
+                        "items": {"type": "string"}
+                    }
```

Comment:
```suggestion
                    "predecessor": {
                          "type": "object",
                          "properties": {
                              "event_id": {"type": "string"},
                              "room_id": {"type": "string", "format": "room_id"}
                          },
                          "required": [
                              "event_id",
                              "room_id"
                          ]
                    }
```

Based on https://spec.matrix.org/v1.1/client-server-api/#mroomcreate this feels like the right thing?